control/tests/discrete_test.py

```diff
@@ -123,41 +130,18 @@ def test_timebase_conversions(self):
         self.assertEqual(timebase(tf3*tf3), timebase(tf3))
         self.assertEqual(timebase(tf3*tf4), timebase(tf4))
         self.assertEqual(timebase(tf3+tf3), timebase(tf3))
-        self.assertEqual(timebase(tf3+tf3), timebase(tf4))
+        self.assertEqual(timebase(tf3+tf4), timebase(tf4))
         self.assertEqual(timebase(feedback(tf3, tf3)), timebase(tf3))
         self.assertEqual(timebase(feedback(tf3, tf4)), timebase(tf4))
 
         # Make sure all other combinations are errors
-        try:
-            tf2*tf3             # Error; incompatible timebases
-            raise ValueError("incompatible operation allowed")
-        except ValueError:
-            pass
-        try:
-            tf2*tf4             # Error; incompatible timebases
-            raise ValueError("incompatible operation allowed")
-        except ValueError:
-            pass
-        try:
-            tf2+tf3             # Error; incompatible timebases
-            raise ValueError("incompatible operation allowed")
-        except ValueError:
-            pass
-        try:
-            tf2+tf4             # Error; incompatible timebases
-            raise ValueError("incompatible operation allowed")
-        except ValueError:
-            pass
-        try:
-            feedback(tf2, tf3)  # Error; incompatible timebases
-            raise ValueError("incompatible operation allowed")
-        except ValueError:
-            pass
-        try:
-            feedback(tf2, tf4)   # Error; incompatible timebases
-            raise ValueError("incompatible operation allowed")
-        except ValueError:
-            pass
+        self.assertRaises(ValueError, tf2.__mul__, tf3)
+        self.assertRaises(ValueError, tf2.__mul__, tf4)
+        self.assertRaises(ValueError, tf2.__add__, tf3)
+        self.assertRaises(ValueError, tf2.__add__, tf4)
+        self.assertRaises(ValueError, tf2.feedback, tf3)
+        self.assertRaises(ValueError, tf2.feedback, tf4)
```

Comment:
People have been fiddling with `__rmul__` and co. It may be better to use

```python
with self.assertRaises(ValueError):
    tf2*tf3
```

Reply:
Thanks for the kinda words! 

I will give that test  formulation a try and leave in that function with a deprecation warning next time I’m at my computer.